extensions/vscode/src/views/homeView.ts

```diff
@@ -771,62 +786,57 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
     this.requestWebviewSaveSelection();
   }
 
-  private async showSelectConfigForDeployment() {
-    const activeDeployment = this.getActiveContentRecord();
-    if (activeDeployment === undefined) {
+  private async showSelectOrCreateConfigForDeployment(
+    targetContentRecord?: ContentRecord | PreContentRecord,
+    entryPoint?: string,
+  ) {
+    if (!targetContentRecord) {
+      targetContentRecord = this.getActiveContentRecord();
+    }
+    if (targetContentRecord === undefined) {
       console.error(
-        "homeView::showSelectConfigForDeployment: No active deployment.",
+        "homeView::showSelectConfigForDeployment: No target deployment.",
       );
       return;
     }
-    const config = await selectConfig(activeDeployment, Views.HomeView);
+    const config = await selectConfig(
+      targetContentRecord,
+      Views.HomeView,
+      entryPoint,
+    );
     if (config) {
       const api = await useApi();
       const apiRequest = api.contentRecords.patch(
-        activeDeployment.deploymentName,
+        targetContentRecord.deploymentName,
         config.configurationName,
-        { dir: activeDeployment.projectDir },
+        { dir: targetContentRecord.projectDir },
       );
       showProgress("Updating Config", Views.HomeView, apiRequest);
 
       await apiRequest;
-    }
-  }
 
-  private async createConfigForDeployment() {
-    const activeDeployment = this.getActiveContentRecord();
-    if (activeDeployment === undefined) {
-      console.error(
-        "homeView::createConfigForDestination: No active deployment.",
-      );
-      return;
-    }
-    // selectConfig handles create as well
-    const config = await selectConfig(activeDeployment, Views.HomeView);
-    if (config) {
-      const activeContentRecord = this.getActiveContentRecord();
-      if (activeContentRecord === undefined) {
-        console.error(
-          "homeView::showSelectConfigForDeployment: No active deployment.",
-        );
-        return;
-      }
-      const api = await useApi();
-      const apiRequest = api.contentRecords.patch(
-        activeContentRecord.deploymentName,
-        config.configurationName,
-        { dir: activeDeployment.projectDir },
-      );
-      showProgress("Updating Deployment", Views.HomeView, apiRequest);
-
-      await apiRequest;
+      // now select the new, updated or existing deployment
+      const deploymentSelector: DeploymentSelector = {
+        deploymentPath: targetContentRecord.deploymentPath,
+      };
+      this.propagateDeploymentSelection(deploymentSelector);
```

Comment:
Do we need to propagate the selection here? We weren't before, and if I'm understanding this correctly it should already be picked up by the file watchers. I may be missing something else though.

Reply:
No, you are correct, they will be set by the state of the files. This is, as you suggest, a much cleaner way to have it happen.